### PR TITLE
[rabit harden] remove rabit_num_trail from dmlc-core

### PR DESCRIPTION
**Pull request by chenqin**
Follow up refactor based on comments after https://github.com/dmlc/dmlc-core/pull/510. 

- Introduce **--local-num-attempt** to config total number of times local tracker could restart workers.(for obvious reasons, this config serves little meaning in distributed environment)
- Pass updated **DMLC_NUM_ATTEMPT** value to mocked rabit worker as number of failed tails local tracker executed in current job.

This pr served a prerequisite of reenable all rabit C++ tests (following pr here https://github.com/dmlc/rabit/pull/81)

### CODE REVIEW

File: tracker/dmlc_tracker/local.py [hunk 1]
```diff
@@ -43,7 +46,7 @@ def exec_cmd(cmd, role, taskid, pass_env):
             if num_retry >= 0:
                 # failure trail increase by 1 and restart failed worker
                 for arg in cmd:
-                    if arg.startswith('rabit_num_trial'):
+                    if arg.startswith('DMLC_NUM_ATTEMPT'):
```
Comment (CodingCat) [inline tracker/dmlc_tracker/local.py]: it's very weird to have this in local.py, this is essentially a param only for test, and particularly only for any test requiring to indicate how many times the process has been started

Comment (CodingCat) [reply tracker/dmlc_tracker/local.py]: the expected solution is to make dmlc-submit as spark-submit, it takes several parameters for its own, like --num-executors, and anything following the path to a jar should be taken as the application-specific params, and they are delegate to the application itself to process

**Comment by chenqin** [comment]
@CodingCat @hcho3

**Comment by chenqin** [comment]
passed xgb integration tests
https://travis-ci.org/chenqin/xgboost/builds/505366624

File: tracker/dmlc_tracker/local.py [hunk 2]
```diff
@@ -21,12 +21,11 @@ def exec_cmd(cmd, role, taskid, pass_env):
...
     env['DMLC_JOB_CLUSTER'] = 'local'
-    num_retry = env.get('DMLC_NUM_ATTEMPT', 0)
 
-    #overwrite default num of retry with commandline value
-    for param in cmd:
-        if param.startswith('DMLC_NUM_ATTEMPT'):
-            num_retry = int(param.split('=')[1])
+    # backward compatibility
+    num_retry = env.get('DMLC_NUM_ATTEMPT', num_attempt)
+    num_trail = 0
```
Comment (szha) [inline tracker/dmlc_tracker/local.py]: trial?

Comment (chenqin) [reply tracker/dmlc_tracker/local.py]: fixed.

**Comment by chenqin** [comment]
^ @szha can we merge this to master https://github.com/dmlc/rabit/issues/82 is point to this fix.

**Comment by chenqin** [comment]
Thanks!

**Pull request merged**